docker-for-mac/networking.md

```diff
@@ -39,9 +39,30 @@ $ docker run -p 8000:80 -d nginx
 Now, connections to `localhost:8000` are sent to port 80 in the container. The
 syntax for `-p` is `HOST_PORT:CLIENT_PORT`.
 
+### SSH agent forwarding
+
+Docker Desktop for {{Arch}} allows you to use the host’s SSH agent inside a container. To do this:
+
+1. Bind mount the SSH agent socket by adding the following parameter to your `docker run` command:
+
+    `-v /run/host-services/ssh-auth.sock:/run/host-services/ssh-auth.sock`
```

Comment:
Although this notation is "easier", the shorthand `-v <host-path>:<container-path>` has known problems with race-conditions, where `<host-path>` does not (yet) exist, it would create a _directory_ with the given name.

Not sure if it applies here, but the long form syntax does not have this problem (it would produce an _error_ instead of silently creating the path).

Should we promote using the long-hand syntax? (see https://docs.docker.com/storage/bind-mounts/#choose-the--v-or---mount-flag)

```bash
--mount type=bind,src=/run/host-services/ssh-auth.sock,target=/run/host-services/ssh-auth.sock
```

Reply:
I didn't know that! Thanks. 

The `/run/host-services/ssh-auth.sock` path is always present. It is created at VM startup time before dockerd boot. I will add a ticket to double-check if a race condition can happen.

Reply:
The 'auto creating host paths' was a "feature" (but causing more troubles than benefits); we tried deprecating/removing it, but some people rely on it. That's why we picked the new (`--mount`) syntax to "break" the feature (but continue having the feature for the "old" `-v foo:bar` shorthand).

It's mostly problematic when bind-mounting _files_ (bind-mounting files is troublesome in all cases, because they are based on inodes, not the _path_, so editing the file could mean that a _new_ file is created at the location, but the container still mounting the old file).

In this case, one possible approach could be to mount the _directory_ containing the socket, so

```bash
-v /run/host-services/:/run/host-services/
```

if `/run/host-services` is a regular directory inside the VM (so not a mount on its own), then worst case, the directory is created when the container is started (but the socket not being there yet)

Reply:
Updated the example to use --mount notation